services/yaml/yaml.tester.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const Joi = require('joi');
+const ServiceTester = require('../service-tester');
+const colorscheme = require('../../lib/colorscheme.json');
+const mapValues = require('lodash.mapvalues');
+
+const colorsB = mapValues(colorscheme, 'colorB');
+
+const t = new ServiceTester({ id: 'dynamic-yaml', title: 'User Defined YAML Source Data', pathPrefix: '/badge/dynamic/yaml' });
+module.exports = t;
+
+t.create('Connection error')
+  .get('.json?url=https://github.com/badges/shields/raw/master/package.json&query=$.name&label=Package Name&style=_shields_test')
+  .networkOff()
+  .expectJSON({ name: 'Package Name', value: 'inaccessible', colorB: colorsB.red });
+
+t.create('No URL specified')
+  .get('.json?query=$.name&label=Package Name&style=_shields_test')
+  .expectJSON({ name: 'Package Name', value: 'no url specified', colorB: colorsB.red });
+
+t.create('No query specified')
+  .get('.json?url=https://raw.githubusercontent.com/kubernetes/charts/568291d6e476c39ca8322c30c3f601d0383d4760/stable/coredns/Chart.yaml&label=Package Name&style=_shields_test')
+  .expectJSON({ name: 'Package Name', value: 'no query specified', colorB: colorsB.red });
+
+t.create('YAML from url')
+  .get('.json?url=https://raw.githubusercontent.com/kubernetes/charts/568291d6e476c39ca8322c30c3f601d0383d4760/stable/coredns/Chart.yaml&query=$.name&style=_shields_test')
+  .expectJSON({ name: 'custom badge', value: 'coredns', colorB: colorsB.brightgreen });
+
+t.create('YAML from uri (support uri query paramater)')
+  .get('.json?uri=https://raw.githubusercontent.com/kubernetes/charts/568291d6e476c39ca8322c30c3f601d0383d4760/stable/coredns/Chart.yaml&query=$.name&style=_shields_test')
+  .expectJSON({ name: 'custom badge', value: 'coredns', colorB: colorsB.brightgreen });
+
+t.create('YAML from url | multiple results')
+  .get('.json?url=https://raw.githubusercontent.com/kubernetes/charts/568291d6e476c39ca8322c30c3f601d0383d4760/stable/coredns/Chart.yaml&query=$..keywords[0:2:1]')
+  .expectJSON({ name: 'custom badge', value: 'coredns, dns' });
+
+t.create('YAML from url | caching with new query params')
+  .get('.json?url=https://raw.githubusercontent.com/kubernetes/charts/568291d6e476c39ca8322c30c3f601d0383d4760/stable/coredns/Chart.yaml&query=$.version')
+  .expectJSONTypes(Joi.object().keys({
+    name: 'custom badge',
+    value: Joi.string().regex(/^\d+(\.\d+)?(\.\d+)?$/)
```

Comment:
I think you can use an `isSemver` validator here: https://github.com/badges/shields/blob/66c5d914c73f39acd6bf44e3d69dffa2c17a20c8/services/xml/xml.tester.js#L55-L57 
Or even better you can change expect to 
```js
.expectJSON({ name: 'custom badge', value: '0.8.0' });
```
since we are using resource from a specific commit.

Reply:
I just copied the test from the json dynamic badge. Happy to update if you prefer.